birdyboard.py

```diff
@@ -0,0 +1,443 @@
+
+    def new_chirp_thread_menu(self):
+
+        print("'b' to go back.\n'x' to exit.\n******NEW " + self.public_or_private + " CHIRP THREAD:******")
+        if chirp_to_add == "b":
+            print("going back.")
+        elif chirp_to_add == "x":
+            print("goodbye.")
+            exit()
+        else:
+            if self.public_or_private == "private":
+                self.private_chirp_thread_menu()
+            else:
+                self.add_new_chirp_thread(chirp_to_add)
+                self.view_chirps()
+                self.view_chirps_next_step()
+
+    def private_chirp_thread_menu(self):
+        pass
+
+    def add_new_thread(self, formatted_chirp):
+        """
+        runs inside 'new chirp menu's, both public and private. adds a new chirp to a selected public chirp thread list, using current user name and text passed in from menu. also makes sure the .txt file is as up to date as possible.
+        Arguments: a formatted chirp. for a public chirp, it's simply a 2-item tuple (user_name, text). for a private chirp, it's a dictionary: {users:(user1, user2), "chirps": (user1, text)}
+        """
+        self.deserialize_chirps_library()
+        self.chirps_library[self.public_or_private][self.current_chirp["index"]].append(formatted_chirp)
+        self.serialize_chirps_library()
+
+# ###############################
+# ######## SERIALIZATION ########
+# ###############################
+
+    def deserialize_chirps_library(self):
+        """
+        opens chirps.txt file. handles what happens if there are no chirps.
+        """
+        try:
+            with open('chirps.txt', 'rb') as chirps:
+                self.chirps_library = pickle.load(chirps)
+
+        except FileNotFoundError:
+                self.chirps_library = {"public": [], "private": []}
+        except EOFError:
+                self.chirps_library = {"public": [], "private": []}
+
+    def serialize_chirps_library(self):
+        """
+        saves new chirps to the chirps.txt file.
+        Arguments: none
+        """
+        with open("chirps.txt", "wb+") as chirps:
+            pickle.dump(self.chirps_library, chirps)
+
+    def deserialize_users(self):
+        """
+        opens users.txt file. handles what happens if there are no users.
+        """
+        try:
+            with open("users.txt", "rb") as users:
+                self.users = pickle.load(users)
+        except FileNotFoundError:
+                self.users = [{"user_name": "megan", "password": "1234"}]
+        except EOFError:
+                self.users = [{"user_name": "megan", "password": "1234"}]
+
+    def serialize_users(self):
+        """
+        saves new users to the users.txt file.
+        Arguments: none
+        """
+        with open("users.txt", "wb+") as users:
+            pickle.dump(self.users, users)
+
+
+if __name__ == '__main__':
+    app = Birdyboard()
+    app.unlogged_in_menu_print()
+    app.unlogged_in_menu_next_step()
```

Comment:
Since you're calling these both in immediate succession, and the `unlogged_in_menu_print()` method only prints, I'd probably take those prints and just tack them on to the front of the `unlogged_in_menu_next_step()` method, since that's essentially what's happening anyway.